### ISSUE TITLE
mempool wont sync

### ISSUE THREAD

**Issue by minioff** [cculianu/Fulcrum#172]
In certain conditions I are observing mempool fails to sync from remote bitcoind. The setup is

bitcoind on machine A
fulcrum on machine A (connected to local bitcoind on A)
fulcrum on machine B (connected to remote bitcoind on A)

- The issue can only be observed on fulcrum running on remote instance B.
- Machine B is located on the other side of the globe about 200ms apart. Both machines are in data centers.
- Both fulcrum are running same versions and configurations. 

The conditions to observe failure are a large mempool (eg 500mb) and somewhat slow connection to remote bitcoind.

**Working situation with mempool decreased to 50mb I can observe** 

```
<Controller> 227 mempool txs involving 1162 addresses
<SynchMempool> (Debug) downloaded 292 txs (failed: 0, ignored: 0), elapsed so far: 43.480 secs
<SynchMempool> (Debug) Mempool: pre-cache of 405 confirmed spends from db took 19.351 msec
<SynchMempool> (Debug) 519 mempool txs involving 1852 addresses (exclusive lock held for 1.859 msec)
<Controller> (Debug) SynchMempool elapsed total: 43.502 secs
```

**With mempool at 500 mb I only get this line**

```
<SynchMempool> (Debug) getrawmempool: got reply with XXXXXXX items, 0 ignored, 0 dropped, XXXXXXX new
```

No further output regarding mempool operations is seen in the log. Mempool always remains empty.

```
./FulcrumAdmin -p 8000 getinfo | jq .mempool
{
  "addresses": 0,
  "avg_fee_sats_B": 0,
  "size_bytes": 0,
  "txs": 0
}
```

**Comment by minioff** [comment]
I have been trying to play with high values (above 100) for `bd-clients` in fulcrum and `rpcthreads` in bitcoind respectively, but it doesn't seem to be making a difference. Can see in bitcoind log with `-debug=rpc` that the throughput is the issue. Is `getrawtransaction` call parallelized?

**Comment by minioff** [comment]
Thanks for confirming my assumptions and observations.

Parallel getrawtx would indeed solve this issue as the machines can easily support 100+ threads.

**Comment by cculianu** [comment]
Yeah it probably would, esp with a 100x speedup.  I guess I just don't have the resources right now to devote to working on this.  It is solvable, for sure, with parallelism. It would be quite an effort and for a niche case, for basically a "free" open source app, kind of too much work for me to do. :)
